include/deal.II/fe/fe_base.h

```diff
@@ -237,12 +274,27 @@ class FiniteElementData
    */
   const unsigned int dofs_per_line;
 
+private:
+  /**
+   * Number of degrees of freedom on quads. If all quads have the same
+   * number of degrees freedoms the values equal dofs_per_quad.
+   */
+  const std::vector<unsigned int> n_dofs_on_quad;
+
+public:
   /**
    * Number of degrees of freedom in a quadrilateral; not including the
    * degrees of freedom on the lines and vertices of the quadrilateral.
    */
   const unsigned int dofs_per_quad;
 
+private:
+  /**
+   * Maximum number of degrees of freedom on any quad.
+   */
+  const unsigned int dofs_per_quad_max;
```

Comment:
`dofs_per_quad_max`, `dofs_per_face_max`, `max_dofs_per_quad()` and `max_dofs_per_face` are part of #10777.